packages/scaffold-ui/src/modal/w3m-modal/index.ts

```diff
@@ -169,18 +167,20 @@ export class W3mModal extends LitElement {
       return
     }
 
-    const previousAddress = CoreHelperUtil.getPlainAddress(this.caipAddress)
     const newAddress = CoreHelperUtil.getPlainAddress(caipAddress)
-    const previousNetworkId = CoreHelperUtil.getNetworkId(this.caipAddress)
     const newNetworkId = CoreHelperUtil.getNetworkId(caipAddress)
-    this.caipAddress = caipAddress
 
     if (this.isSiweEnabled) {
       const { SIWEController } = await import('@web3modal/siwe')
       const session = await SIWEController.getSession()
 
+      // Noop if there is no session
+      if (!session?.address || !session?.chainId) {
```

Comment:
@Cali93 i think this will cause issues. If the user connects to a wallet and refreshes the dapp before signing, the sign popup won't appear again

Reply:
https://github.com/user-attachments/assets/9b943f48-9dea-4686-a8a4-6f1ec1e1f113